api/tests/authentication_tests.py

```diff
@@ -18,6 +32,213 @@ def _setup_fxa_response(status_code: int, json: dict | str):
     return {"status_code": status_code, "json": json}
 
 
+def _setup_fxa_response_request_exception(status_code: int, json: dict | str):
+    fxa_verify_path = "https://oauth.stage.mozaws.net/v1/introspect"
+    resp = responses.Response(
+        method="POST",
+        url=fxa_verify_path,
+        body=Timeout(),
+        status=status_code,
+        json=json,
+    )
+    responses.add(resp)
+    return {"status_code": status_code}
+
+
+def _setup_fxa_response_no_json(status_code: int):
+    fxa_verify_path = "https://oauth.stage.mozaws.net/v1/introspect"
+    responses.add(responses.POST, fxa_verify_path, status=status_code)
+    return {"status_code": status_code}
+
+
+class AuthenticationMiscellaneous(TestCase):
+    def setUp(self):
+        self.auth = FxaTokenAuthentication
+        self.factory = RequestFactory()
+        self.path = "/api/v1/relayaddresses"
+        self.fxa_verify_path = "https://oauth.stage.mozaws.net/v1/introspect"
+        self.uid = "relay-user-fxa-uid"
+
+    def tearDown(self):
+        cache.clear()
+
+    @responses.activate()
+    def test_introspect_token_catches_JSONDecodeError_raises_AuthenticationFailed(self):
+        _setup_fxa_response_no_json(200)
+        invalid_token = "invalid-123"
+        cache_key = get_cache_key(invalid_token)
+
+        try:
+            introspect_token(cache_key, invalid_token)
+        except AuthenticationFailed as e:
+            assert str(e.detail) == "JSONDecodeError from FXA introspect response"
+            assert responses.assert_call_count(self.fxa_verify_path, 1) is True
+            return
+        self.fail("Should have raised AuthenticationFailed")
+
+    @responses.activate()
+    def test_introspect_token_returns_fxa_introspect_response(self):
+        now_time = int(datetime.now().timestamp())
+        # Note: FXA iat and exp are timestamps in *milliseconds*
+        exp_time = (now_time + 60 * 60) * 1000
+        json_data = {"active": True, "sub": self.uid, "exp": exp_time}
+        status_code = 200
+        expected_fxa_resp_data = {"status_code": status_code, "json": json_data}
```

Comment:
Nit: This code appears repeated a few time, could be a helper function

Reply:
Created a separate ticket to do test clean-up: https://mozilla-hub.atlassian.net/browse/MPP-3162